src/utils/hibp.js

```diff
@@ -0,0 +1,205 @@
+import got from 'got'
+import mozlog from './log.js'
+import AppConstants from '../app-constants.js'
+import { fluentError } from './fluent.js'
+const { HIBP_THROTTLE_MAX_TRIES, HIBP_THROTTLE_DELAY, HIBP_API_ROOT, HIBP_KANON_API_ROOT, HIBP_KANON_API_TOKEN } = AppConstants
+
+// TODO: fix hardcode
+const HIBP_USER_AGENT = 'monitor/1.0.0'
+// When HIBP "re-names" a breach, it keeps its old 'Name' value but gets a new 'Title'
+// We use 'Name' in Firefox (via Remote Settings), so we have to maintain our own mapping of re-named breaches.
+const RENAMED_BREACHES = ['covve']
+const RENAMED_BREACHES_MAP = {
+  covve: 'db8151dd'
+}
+const log = mozlog('hibp')
+
+function _addStandardOptions (options = {}) {
+  const hibpOptions = {
+    headers: {
+      'User-Agent': HIBP_USER_AGENT
+    },
+    responseType: 'json'
+  }
+  return Object.assign(options, hibpOptions)
+}
+
+export async function _throttledGot (url, reqOptions, tryCount = 1) {
```

Comment:
I wonder if throttling is still necessary – might be something to ask @groovecoder and/or Troy@HIBP?  We get a lot of Sentry errors related to HIBP connections, and I wonder if it's because we're hitting max-tries and giving up before it's necessary?

Reply:
It's still necessary. As long as we get 429s from HIBP (which we will, given how many requests we send), we will need some type of throttling (via setTimeOut here). 

The right way to pipe these requests is via a queue, which will take some more engineering efforts. I can create a spike ticket to look into that

Reply:
Re Sentry: 

Right now with Sentry, there are lots of false positives since we aren't configuring it at all. Some errors are actual errors and should be tracked, some are actually not errors, but are just thrown to give some feedback to the front end (like invalid sessions). We definitely need to be more mindful about what goes into Sentry.